Fwk/VizFwk/LibGuiQt/cvfqtGLWidget.cpp

```diff
@@ -0,0 +1,518 @@
+//
+//   GNU General Public License Usage
+//   This library is free software: you can redistribute it and/or modify
+//   it under the terms of the GNU General Public License as published by
+//   the Free Software Foundation, either version 3 of the License, or
+//   (at your option) any later version.
+//
+//   This library is distributed in the hope that it will be useful, but WITHOUT ANY
+//   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//   FITNESS FOR A PARTICULAR PURPOSE.
+//
+//   See the GNU General Public License at <<http://www.gnu.org/licenses/gpl.html>>
+//   for more details.
+//
+//   GNU Lesser General Public License Usage
+//   This library is free software; you can redistribute it and/or modify
+//   it under the terms of the GNU Lesser General Public License as published by
+//   the Free Software Foundation; either version 2.1 of the License, or
+//   (at your option) any later version.
+//
+//   This library is distributed in the hope that it will be useful, but WITHOUT ANY
+//   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//   FITNESS FOR A PARTICULAR PURPOSE.
+//
+//   See the GNU Lesser General Public License at <<http://www.gnu.org/licenses/lgpl-2.1.html>>
+//   for more details.
+//
+//##################################################################################################
+
+#include "cvfqtGLWidget.h"
+
+#include "cvfOpenGLContextGroup.h"
+#include "cvfOpenGLContext.h"
+#include "cvfLogManager.h"
+#include "cvfTrace.h"
+
+#include <QPointer>
+#include <QEvent>
+#include <QOpenGLContext>
+#include <QPlatformSurfaceEvent>
+
+namespace cvfqt {
+
+
+
+//==================================================================================================
+//
+// 
+//
+//==================================================================================================
+class ForwardingOpenGLContext_GLWidget : public cvf::OpenGLContext
+{
+public:
+    ForwardingOpenGLContext_GLWidget(cvf::OpenGLContextGroup* contextGroup, QGLWidget* ownerQtGLWidget)
+    :   cvf::OpenGLContext(contextGroup),
+        m_ownerQtGLWidget(ownerQtGLWidget)
+    {
+        CVF_ASSERT(contextGroup);
+
+        // In our current usage pattern the owner widget (and its contained Qt OpenGL context) must already be initialized/created
+        CVF_ASSERT(m_ownerQtGLWidget);
+        CVF_ASSERT(m_ownerQtGLWidget->isValid());
+        CVF_ASSERT(m_ownerQtGLWidget->context());
+        CVF_ASSERT(m_ownerQtGLWidget->context()->isValid());
+    }
+
+    virtual void makeCurrent()
+    {
+        if (m_ownerQtGLWidget)
+        {
+            m_ownerQtGLWidget->makeCurrent();
+        }
+    }
+    
+    virtual bool isCurrent() const
+    {
+        const QGLContext* ownersQGLContext = m_ownerQtGLWidget ? m_ownerQtGLWidget->context() : NULL;
+        if (ownersQGLContext && QGLContext::currentContext() == ownersQGLContext)
+        {
+            return true;
+        }
+
+        return false;
```

Comment:
Prefer nullptr to NULL?

Reply:
VizFwk is not modernized to use nullptr, will add a separate issue for this.